In [1]:
import matplotlib.pyplot as plt
%matplotlib inline
import numpy as np
import pandas as pd
import statsmodels.api as sm
from statsmodels.nonparametric.kde import KDEUnivariate
from statsmodels.nonparametric import smoothers_lowess
from pandas import Series, DataFrame
from patsy import dmatrices
from sklearn import datasets, svm
import predict as ka # see github.com/agconti/kaggleaux for more details
from sklearn.model_selection import train_test_split
import sklearn.ensemble as ske
import sys

/home/daniel/anaconda3/lib/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


# Collect all data

In [7]:
df_list=[]
for i in range(25):
    df = pd.read_csv("features/{}.csv".format(i),dtype=float)
    df.columns=['f{}'.format(j) for j in range(df.shape[1])]
    df_list.append(df)

df_all=pd.concat(df_list)
print(df_all.shape)

(25000, 4098)


In [3]:
formula = "f1 ~ f2" 
for i in range(3,df_all.shape[1]):
     formula+='+f{}'.format(i)
results = {}

# Training and validation test
### Using Random Forest method

In [8]:
train, test = train_test_split(df_all, test_size = 0.2)

sys.setrecursionlimit(20000)

# I use train data
y, x = dmatrices(formula, data=train, return_type='dataframe')
y = np.asarray(y).ravel()

results_rf = ske.RandomForestClassifier(n_estimators=100).fit(x, y)

# I use test data
yt, xt = dmatrices(formula, data=test, return_type='dataframe')
yt = np.asarray(yt).ravel()


score = results_rf.score(xt, yt)
print("Random Forest Accuracy: {}".format(score))

Random Forest Accuracy: 0.9812


### My model accuracy for Dog vs Cat problem is ~0.98